src/GeekLearning.Storage.FileSystem/FileSystemStore.cs

```diff
@@ -130,21 +199,19 @@ private void EnsurePathExists(string path)
             }
         }
 
-        private Internal.FileSystemFileReference InternalGetAsync(IPrivateFileReference file)
+        private static string GenerateEtag(string fileSystemPath)
         {
-            var reference = InternalGetOrCreateAsync(file);
-            if (File.Exists(reference.FileSystemPath))
+            var data = File.ReadAllBytes(fileSystemPath);
```

Comment:
@asiffermann this is not highest priority and might be tracked as an issue if you don't feel like addressing it now.